src/ServiceManager.php

```diff
@@ -415,8 +404,11 @@ public function setInvokableClass($name, $class = null)
      * Specify a factory for a given service name.
      *
      * @param string $name Service name
-     * @param string|callable|Factory\FactoryInterface $factory Factory to which
-     *     to map.
+     * @param string|callable|Factory\FactoryInterface $factory  Factory to which to map.
+     * phpcs:disable Generic.Files.LineLength.TooLong
+     * @psalm-param class-string<Factory\FactoryInterface>|callable(ContainerInterface,string,ServiceOptionsType|null)|Factory\FactoryInterface $factory
```

Comment:
it seems that `@psalm-param` cannot properly detect multiline declaration and thus, I had to use `phpcs:disable` to disable the line length check. 
Screw you, line length check!